File: back/ingang.py
import json

async def get_token(requests, id, password):
  req = await requests.post('http://dev-api.dimigo.in/auth', json={
    'id': id,
    'password': password
  })
  return json.loads(req.text)['token']

async def get_ingangs(requests, token):
  req = await requests.get('http://dev-api.dimigo.in/ingang', headers={
    'Authorization': 'Bearer {}'.format(token)
  })
  return json.loads(req.text)['ingangs']

async def _request_ingang(requests, token, ingang): 
  req = await requests.post('http://dev-api.dimigo.in/ingang/{}'.format(ingang['idx']), headers={
    'Authorization': 'Bearer {}'.format(token)
  })
  return req.status_code

async def apply_ingang(requests, token, ingang):
  for i in range(100):
    while 1:
      try:
        req_code = await _request_ingang(requests, token, ingang)
        break
      except ConnectionError:
        print('[!] ConnectionError occurred')
    print(req_code)
    if req_code == 200: 
      print('[+] {}번 인강 신청 성공'.format(ingang.idx))
      return 200
    elif req_code == 403:
      print('[!] 모든 티켓을 사용했습니다.')
      return 403
    sleep(1)

File: utils/ingang.js
const axios = require('axios');
const api = axios.create({
  baseURL: 'http://dev-api.dimigo.in/'
});

module.exports = {
  async getToken (id, password) {
    return api.post('/auth', {
      id, password
    })
      .then((res) => {
        return res.data.token
      })
  },

  async getIngangs (token) {
    return api.get('/ingang', {
      headers: {
        Authorization: `Bearer ${token}`
      }
    })
      .then((res) => {
        return res.data.ingangs
      })
  },

  async requestIngang (ingang, token) {
    return api.post(`/ingang/${ingang.idx}`, {}, {
      headers: {
        Authorization: `Bearer ${token}`
      }
    })
      .then((res) => {
        if (res.status === 200)
          return true
        return res.status
      })
      .catch((err) => {
        return err.response.status
      })
  },

  async apply (secret, time) {
    // time is 1 or 2
    return this.getToken(secret.id, secret.password)
      .then((token) => {
        console.log(token)
        return this.getIngangs(token)
          .then((ingangs) => {
            console.log(ingangs)
            return this.requestIngang(ingangs[time - 1], token)
              .then((res) => {
                console.log(res)       
                return res       
              })
          })
      });
  }
};
